sdk/core/azure-core/azure/core/settings.py

```diff
@@ -156,18 +161,19 @@ def convert_tracing_impl(value):
     if value is None:
         return get_opencensus_span_if_opencensus_is_imported()
 
-    wrapper_class = value
-    if isinstance(value, six.string_types):
-        value = value.lower()
-        get_wrapper_class = _tracing_implementation_dict.get(value, lambda: _Unset)
-        wrapper_class = get_wrapper_class()
-        if wrapper_class is _Unset:
-            raise ValueError(
-                "Cannot convert {} to AbstractSpan, valid values are: {}".format(
-                    value, ", ".join(_tracing_implementation_dict)
-                )
+    if not isinstance(value, six.string_types):
+        return value
+
+    value = cast(str, value)  # mypy clarity
+    value = value.lower()
+    get_wrapper_class = _tracing_implementation_dict.get(value, lambda: _Unset)
```

Comment:
if value is an of type AbstractSpan then the dict will return _unset when it should just return itself.

Reply:
Didn't change this code, just indented it differently :)